internal/handler/msg_createuser.go

```diff
@@ -245,5 +146,53 @@ func makeCredentials(mechanisms *types.Array, username, userPassword string) (*t
 		}
 	}
 
-	return credentials, nil
+	if document.Has("pwd") {
+		pwdi := must.NotFail(document.Get("pwd"))
+		userPasswordV, ok := pwdi.(string)
+
+		if !ok {
+			return nil, handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrTypeMismatch,
+				fmt.Sprintf("BSON field 'createUser.pwd' is the wrong type '%s', expected type 'string'",
+					handlerparams.AliasFromType(pwdi),
+				),
+			)
+		}
+
+		if userPasswordV == "" {
+			return nil, handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrSetEmptyPassword,
+				"Password cannot be empty",
+			)
+		}
+
+		prepped, err := stringprep.SASLprep.Prepare(userPasswordV)
+		if err != nil {
+			return nil, handlererrors.NewCommandError(
+				handlererrors.ErrStringProhibited,
+				err,
```

Comment:
@chilagrow error message needs to be equal, see CI [failure](https://github.com/FerretDB/FerretDB/actions/runs/8812334411/job/24187824583?pr=4248#step:10:99). Should I hard code the expected error message?

Reply:
I'm not sure why we are importing `stringprep` here, that logic belongs to password and we shouldn't change that.

Here we should get error back from backend, and handle that error. Not importing other functions.

Reply:
In around L186, once CreateUser is called, check error by using the same logic as before https://github.com/FerretDB/FerretDB/pull/4248/files#diff-9666c62739089acb9bca18ed035a560fba6a086493bde93c1483e356e44d6f54L228-L237